### PR TITLE
Missed account keys generator

### PR DESCRIPTION
**Pull request by r8d8**
Generate secret key and extract from it public key and address.

Related to #29.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -38,6 +38,8 @@ glob = "0.2"
 tempdir = "0.3"
 uuid = { version = "0.4", features = ["rustc-serialize", "v4"] }
 rust-crypto = "0.2"
+secp256k1 = "0.6.0"
+rand = "0.3.15"
```
Comment (dulanov) [inline Cargo.toml]: Probably better to go away from minor version.

File: src/key_generator/mod.rs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+extern crate rand;
+
+use self::rand::{ThreadRng, thread_rng};
+use self::secp256k1::Secp256k1;
+use self::secp256k1::key::SecretKey;
+
+
+/// Errors for key generation.
+pub enum GeneratorError {
+    InvalidKey,
```
Comment (dulanov) [inline src/key_generator/mod.rs]: Not used?

File: src/key_generator/mod.rs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        GeneratorError::InvalidKey
+    }
+}
+
+/// Secret key generator.
+/// Runs for `ticks` times, yielding each time key.
+pub struct Generator<'call> {
+    ticks: u64,
+    secp256: Secp256k1,
+    thread_rng: &'call Fn() -> ThreadRng,
```
Comment (dulanov) [inline src/key_generator/mod.rs]: Really?

File: src/key_generator/mod.rs [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+impl From<secp256k1::Error> for GeneratorError {
+    fn from(err: secp256k1::Error) -> Self {
+        GeneratorError::InvalidKey
+    }
+}
+
+/// Secret key generator.
+/// Runs for `ticks` times, yielding each time key.
+pub struct Generator<'call> {
+    ticks: u64,
```
Comment (dulanov) [inline src/key_generator/mod.rs]: We should start from simple variant, just to generate only one key.

**Comment by dulanov** [comment]
Add also methods to extract public key and address from private key.

**Comment by dulanov** [comment]
@r8d8 As a target result, you need to generate a new keystore file for given passphrase in format `UTC-<timestamp>Z-<uuid>`, additional parameter should decide to open address or not.

File: src/key_generator/mod.rs [hunk 5]
```diff
@@ -1,75 +1,50 @@
...
+use secp256k1::{Error as SecpError, Secp256k1};
 use secp256k1::key::{PublicKey, SecretKey};
 
+
 lazy_static! {
     static ref SECP256K1: Secp256k1 = Secp256k1::new();
 }
 
-/// Errors for key generation.
+/// Errors for key generation
```
Comment (dulanov) [inline src/key_generator/mod.rs]: Respect man ;)

File: src/key_generator/util.rs [hunk 6]
```diff
@@ -0,0 +1,135 @@
+//! # Helper methods for Secret key generator
```
Comment (dulanov) [inline src/key_generator/util.rs]: Util not a good name, really, normally it's a symptom of bad architecture, try to find better decomposition scheme.

Comment (dulanov) [inline src/key_generator/util.rs]: PrivateKey struct probably? or something similar i think

File: src/key_generator/util.rs [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+}
+
+/// Serializes KeyFile into JSON file with name `UTC-<timestamp>Z--<uuid>`
+///
+/// # Arguments
+///
+/// * `kf` - `KeyFile`
+/// * `dir` - path to destination directory
+///
+pub fn to_file(kf: KeyFile, dir: Option<&Path>) -> Result<File, SecpError> {
```
Comment (dulanov) [inline src/key_generator/util.rs]: aaa, it's just serialisation stuff, please don't mix domain logic and IO

File: src/keystore/mod.rs [hunk 8]
```diff
@@ -70,11 +70,13 @@ pub struct KeyFile {
...
 
 impl KeyFile {
-    fn new() -> Self {
+    ///
+    pub fn new() -> Self {
         Self::from(Uuid::new_v4())
     }
 
-    fn with_address(&mut self, addr: &Address) {
+    ///
```
Comment (dulanov) [inline src/keystore/mod.rs]: ???

Comment (r8d8) [inline src/keystore/mod.rs]: Formatter added by default.

File: src/keystore/extract_key.rs [hunk 9]
```diff
@@ -29,6 +29,10 @@ impl KeyFile {
     /// Create keystore file from private key with a passphrase
     pub fn insert_key(&mut self, pk: &[u8], passphrase: &str) -> Result<()> {
         let derived = derive_key(self.dk_length, self.kdf, &self.kdf_salt, passphrase);
+
+        use rustc_serialize::hex::{FromHex, ToHex};
```
Comment (dulanov) [inline src/keystore/extract_key.rs]: To upper level please

File: src/keystore/extract_key.rs [hunk 10]
```diff
@@ -29,6 +29,10 @@ impl KeyFile {
...
+        println!("DEBUG derived key: {:?}", derived.to_hex());
```
Comment (dulanov) [inline src/keystore/extract_key.rs]: Use logs instead or for debugging purposes only!

**Pull request merged**